### PR TITLE
Secure the maker's API routes with basic authentication

### LINKED ISSUE #76: Guard the maker's interface and API with authentication
We will initially host the broker with a publicly accessible IP. We need to guard the API and frontend with some authentication to make sure no one can muck around with our instance.

### PR DESCRIPTION
**Pull request by thomaseizinger**
Resolves #76.

We choose basic authentication because it is simple.
Basic auth allows us to not code any particular login API and instead just trigger the browser's built-in authentication dialog via the `WWW-Authenticate` header of the 401 Unauthorized response. It looks like this:

image

The password is derived from the maker's seed.
The username is always `maker`.

The implementation has one ceveat: Basic authentication is vulnerable to MITM attacks unless some form of transport encryption is in place. I suggest all securing the maker's API additionally with TLS. However, I think we can easily ship this as a feature on top!

### CODE REVIEW

File: daemon/src/routes_maker.rs [hunk 1]
```diff
@@ -103,3 +124,71 @@ pub async fn post_sell_order(
...
+            .manage(state2)
+            .manage(state3)
+            .manage(state4)
+            .manage(Password::from(*b"Now I'm feelin' so fly like a G6"))
+    }
+
+    /// Creates an "Authorization" header that matches the password above,
+    /// in particular it has been created through:
+    /// ```
+    /// base64(maker:hex("Now I'm feelin' so fly like a G6"))
```
Comment (bonomat) [inline daemon/src/routes_maker.rs:228]: 😅

**Review by bonomat — APPROVED** [review]
Would it not make sense to also have authentication for the taker?

File: daemon/src/routes_maker.rs (hunk 1 above)
Comment (thomaseizinger) [reply daemon/src/routes_maker.rs:228]: I was listening to music and needed some random 32 bytes ...

**Comment by thomaseizinger** [comment]
> Would it not make sense to also have authentication for the taker? 

We can trivially reuse this for the taker as well. I thought the taker is most likely run locally and on-demand whereas the maker is probably on a server somewhere due to the need for an open port!

**Review by da-kami — APPROVED** [review]

**Review by klochowicz — APPROVED** [review]
nice one!

**Pull request merged**